File: dot_local/share/codex/skills/sem/references/sem/crates/sem-core/tests/fixtures/scope_test/java/Service.java
public class Service {
    public Dog createDog(String name) {
        Dog dog = new Dog(name);
        if (!dog.validate()) {
            throw new IllegalArgumentException("invalid dog");
        }
        Connection conn = DatabaseHelper.getConnection();
        conn.execute("INSERT INTO dogs VALUES (?)");
        conn.commit();
        return dog;
    }

    public Cat createCat(String name) {
        Cat cat = new Cat(name);
        if (!cat.validate()) {
            throw new IllegalArgumentException("invalid cat");
        }
        Connection conn = DatabaseHelper.getConnection();
        conn.execute("INSERT INTO cats VALUES (?)");
        conn.commit();
        return cat;
    }

    public void transferAnimal(Object animal, Shelter shelter) {
        Transaction txn = new Transaction(DatabaseHelper.getConnection());
        txn.execute("UPDATE animals SET shelter_id = ?");
        shelter.add(animal);
        txn.commit();
    }

    public Object listAnimals() {
        Connection conn = DatabaseHelper.getConnection();
        return conn.execute("SELECT * FROM animals");
    }
}

File: dot_local/share/codex/skills/sem/references/sem/crates/sem-core/tests/fixtures/scope_test/csharp/Service.cs
public class Service {
    public Dog CreateDog(string name) {
        Dog dog = new Dog(name);
        if (!dog.Validate()) {
            throw new ArgumentException("invalid dog");
        }
        Connection conn = DatabaseHelper.GetConnection();
        conn.Execute("INSERT INTO dogs VALUES (?)");
        conn.Commit();
        return dog;
    }

    public Cat CreateCat(string name) {
        Cat cat = new Cat(name);
        if (!cat.Validate()) {
            throw new ArgumentException("invalid cat");
        }
        Connection conn = DatabaseHelper.GetConnection();
        conn.Execute("INSERT INTO cats VALUES (?)");
        conn.Commit();
        return cat;
    }

    public void TransferAnimal(object animal, Shelter shelter) {
        Transaction txn = new Transaction(DatabaseHelper.GetConnection());
        txn.Execute("UPDATE animals SET shelter_id = ?");
        shelter.Add(animal);
        txn.Commit();
    }

    public object ListAnimals() {
        Connection conn = DatabaseHelper.GetConnection();
        return conn.Execute("SELECT * FROM animals");
    }
}
